framework/OpenMod.Core/Users/Events/IUserBanningEvent.cs

```diff
@@ -0,0 +1,23 @@
+﻿using OpenMod.API.Eventing;
+using System;
+
+namespace OpenMod.Core.Users.Events
+{
+    public interface IUserBanningEvent : IUserEvent, ICancellableEvent
+    {
+        /// <summary>
+        /// Gets the punisher Id.
+        /// </summary>
+        public string PunisherId { get; }

```

Comment:
Add `PunisherType`

Reply:
Why punisher type you mean IUser and the punisher id?